library/proc_macro/src/diagnostic.rs

```diff
@@ -1,3 +1,7 @@
+//! This module provides the ability to emit compiler errors to macro developers. Instead of
+//! using `panic!()` and letting the macro panic, you can use the [`Diagnostic`] structure for more helpful
```

Comment:
It's probably worth mentioning `compile_error!` here, which is the current stable recommended way to do errors.